src/libclient/measurement/ping/ping_linux.cpp

```diff
@@ -252,10 +252,28 @@ Result Ping::result() const
     QVariantList roundTripMs;
     float avg = 0.0;
     QList<float> tmpPingTime = pingTime;
+    quint32 tmpPingsReceived = m_pingsReceived;
+    quint32 tmpPingsSent = m_pingsSent;
 
     // calculate average and fill ping times
     foreach (float val, tmpPingTime)
     {
+        // RTTs greater than receiveTimeout are discarded and don't influence the result
+        if (val >= definition->receiveTimeout)
+        {
+            if (tmpPingsReceived > 0)
```

Comment:
I understand the TCP part looks confusing but I checked the code paths as well as I could, and there shouldn't be a problem regarding `m_pingsSent` and `m_pingsReceived`.